src/coprocessor/statistics/analyze.rs

```diff
@@ -715,10 +715,18 @@ impl RowSampleCollector for ReservoirRowSampleCollector {
         let cur_rng = self.base.rng.gen_range(0, i64::MAX);
         if self.samples.len() < self.max_sample_size {
             need_push = true;
-        } else if self.samples.peek().unwrap().0.0 < cur_rng {
-            need_push = true;
-            let (_, evicted) = self.samples.pop().unwrap().0;
-            self.base.memory_usage -= evicted.iter().map(|x| x.capacity()).sum::<usize>();
+        } else {
```

Comment:
I think this can not solve the upgrade consistency problem. it will run sample branch if sample size=1 in master

Reply:
but according to https://github.com/tikv/tikv/issues/11295, samples will be None in some cases